packages/nimble-components/src/spinner/styles.ts

```diff
@@ -0,0 +1,216 @@
+import { css } from '@microsoft/fast-element';
+import { display } from '@microsoft/fast-foundation';
+import {
+    Black15,
+    Black91,
+    White
+} from '@ni/nimble-tokens/dist/styledictionary/js/tokens';
+import { Theme } from '../theme-provider/types';
+import { hexToRgbaCssColor } from '../utilities/style/colors';
+import { themeBehavior } from '../utilities/style/theme';
+
+export const styles = css`
+    ${display('inline-block')}
+
+    :host {
+        width: 32px;
+        height: 32px;
+    }
+
+    :host([size='small']) {
+        width: 16px;
+        height: 16px;
+    }
+
+    :host([size='medium-large']) {
+        width: 48px;
+        height: 48px;
+    }
+
+    :host([size='large']) {
+        width: 64px;
+        height: 64px;
+    }
+
+    :host([size='x-large']) {
+        width: 96px;
+        height: 96px;
+    }
+
+    :host([size='xx-large']) {
+        width: 128px;
+        height: 128px;
+    }
+
+    div.container {
+        width: 87.5%;
+        height: 87.5%;
+        top: 6.25%;
+        left: 6.25%;
+        position: relative;
+    }
+
+    :host([size='small']) div.container {
+        width: 12px;
+        height: 12px;
+        top: 2px;
+        left: 2px;
+    }
+
+    :host([size='medium-large']) div.container {
+        width: 40px;
+        height: 40px;
+        top: 4px;
+        left: 4px;
+    }
+
+    :host([size='x-large']) div.container {
+        width: 80px;
+        height: 80px;
+        top: 8px;
+        left: 8px;
+    }
+
```

Comment:
is the idea that for large and xx-large that the equations work out exactly or something?

Reply:
Yep. With the powers of 2 (32px/64px/128px) the equation yields exact integer values.
The 48px/96px sizes have slightly different padding so they're specified directly. (Also the 'equation' for them would be 5/6 and 1/12, and I was seeing non-integer offsets in the Chrome if I calculated these (due to the .08333... .8333... repeating values they represent I guess). So I just wrote them out fully.